src/test/scala/mesosphere/marathon/core/health/impl/HealthCheckWorkerActorTest.scala

```diff
@@ -83,5 +87,41 @@ class HealthCheckWorkerActorTest extends AkkaUnitTest with ImplicitSender {
       watch(ref)
       expectTerminated(ref)
     }
+
+    "A HTTP health check should work as expected" in {
+
+      import akka.http.scaladsl.server.Directives._
+
+      val promise = Promise[String]()
+
+      val route =
+        path("health") {
+          get {
+            promise.success("success")
+            complete(StatusCodes.OK)
+          }
+        }
+
+      val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+
+      val hostName = "localhost"
+      val appId = PathId("/test_id")
+      val app = AppDefinition(id = appId, portDefinitions = Seq(PortDefinition(0)))
+      val agentInfo = AgentInfo(host = hostName, agentId = Some("agent"), region = None, zone = None, attributes = Nil)
+      val task = {
+        val t: Task = TestTaskBuilder.Helper.runningTaskForApp(appId)
+        val hostPorts = Seq(8080)
+        t.copy(status = t.status.copy(networkInfo = NetworkInfo(hostName, hostPorts, ipAddresses = Nil)))
+      }
+      val instance = LegacyAppInstance(task, agentInfo, UnreachableStrategy.default())
```

Comment:
we should try to avoid using the old `LegacyAppInstance`.

Reply:
sure